lib/integrations/awsoidc/eks_list_clusters.go

```diff
@@ -0,0 +1,186 @@
+	// If empty, the first page is fetched.
+	NextToken string
+}
+
+// CheckAndSetDefaults checks if the required fields are present.
+func (req *ListEKSClustersRequest) CheckAndSetDefaults() error {
+	if req.Region == "" {
+		return trace.BadParameter("region is required")
+	}
+
+	return nil
+}
+
+// EKSCluster represents a cluster in AWS EKS.
+type EKSCluster struct {
+	// Name is the name of AWS EKS cluster.
+	Name string
+
+	// Region is an AWS region.
+	Region string
+
+	// Labels are labels of a EKS cluster.
+	Labels map[string]string
+
+	// ExtraLabels are additional Teleport labels that should be injected into kube agent
+	// if the cluster will be enrolled into Teleport (agent installed on it).
+	ExtraLabels map[string]string
+
+	// Status is a current status of an EKS cluster in AWS.
+	Status string
+}
+
+// ListEKSClustersResponse contains a page of AWS EKS Clusters.
+type ListEKSClustersResponse struct {
+	// Servers contains the page of Servers.
+	Clusters []EKSCluster
+
+	// NextToken is used for pagination.
+	// If non-empty, it can be used to request the next page.
+	NextToken string
+
+	// ClusterFetchingErrors contains errors for fetching detailed information about specific cluster, if any happened.
+	ClusterFetchingErrors map[string]error
+}
+
+// NewListEKSClustersClient creates a new ListEKSClusters client using AWSClientRequest.
+func NewListEKSClustersClient(ctx context.Context, req *AWSClientRequest) (ListEKSClustersClient, error) {
+	clt, err := newEKSClient(ctx, req)
+	return clt, trace.Wrap(err)
+}
+
+// ListEKSClustersClient describes the required methods to List EKS clusters using a 3rd Party API.
+type ListEKSClustersClient interface {
+	// ListClusters lists the EKS clusters.
+	ListClusters(ctx context.Context, params *eks.ListClustersInput, optFns ...func(*eks.Options)) (*eks.ListClustersOutput, error)
+
+	// DescribeCluster returns detailed information about an EKS cluster.
+	DescribeCluster(ctx context.Context, params *eks.DescribeClusterInput, optFns ...func(*eks.Options)) (*eks.DescribeClusterOutput, error)
+}
+
+// concurrentEKSFetchingLimit is a limit of how many clusters we are trying to describe concurrently, after receiving a list of clusters.
+const concurrentEKSFetchingLimit = 5
+
+// ListEKSClusters calls the following AWS API:
+// https://docs.aws.amazon.com/eks/latest/APIReference/API_ListClusters.html - to list available EKS clusters
+// https://docs.aws.amazon.com/eks/latest/APIReference/API_DescribeCluster.html - to get more detailed information about
+// the each cluster in the list we received.
+// It returns a list of EKS clusters with detailed information about them.
+func ListEKSClusters(ctx context.Context, clt ListEKSClustersClient, req ListEKSClustersRequest) (*ListEKSClustersResponse, error) {
+	if err := req.CheckAndSetDefaults(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	listEKSClusters := &eks.ListClustersInput{
+		Include: nil,
```

Comment:
Do we need to initialize the `Include` value?
From docs only blank and "all" are supported values.

If we really need to initialize this to `nil`, can we please add a comment?